packages/storybook/.storybook/webpack-plugin-fail-build-on-warning.js

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+class WebpackPluginFailBuildOnWarning {
```

Comment:
I think we want to add a comment a bit of context and possibly alternative way of doing this since it's pretty custom 😁 

Explaining why we want to fail on warnings and why this webpack build is different (ts-loader with transpileOnly). Also noting that if there's a simpler solution we could go with that instead.

Reply:
@Rugvip Added comment, thank you for the review. Please take a look. :)